sbi/inference/posteriors/direct_posterior.py

```diff
@@ -101,17 +102,13 @@ def sample(
         """
 
         num_samples = torch.Size(sample_shape).numel()
-        condition_shape = self.posterior_estimator._condition_shape
         x = self._x_else_default_x(x)
 
-        try:
-            x = x.reshape(*condition_shape)
-        except RuntimeError as err:
-            raise ValueError(
-                f"Expected a single `x` which should broadcastable to shape \
-                  {condition_shape}, but got {x.shape}. For batched eval \
-                  see issue #990"
-            ) from err
+        # [1:] because we remove batch dimension for `reshape_to_batch_event`.
+        # Note: This line will break if `x_shape` is `None` and if `x` is passed without
```

Comment:
`x_shape` should really only contain the `event_shape`, no? (at least in the future, I think currently it still (can) contain the batch dims).

This will also likely break if xs is a 2d image passed without a batch_dimension.

We should also write an informative error here if something unexpected is passed ( + issue for implementing x_shape standard).

Or ensure that `x_shape` will always contain the `event_shape`

Reply:
I did not want to change `posterior._x_shape` and `inference._x_shape`. Therefore, right now, these contain also the batch dimension. I do, however, agree that we should change this in the future.

In general, if the user uses the `flexible interface`, then `posterior._x_shape` will not be `None` and there is no problem, even for images.

Reply:
I see, okay, then, let's leave it as it is for now.